jdbc/runtime/src/main/java/io/quarkiverse/config/jdbc/runtime/Repository.java

```diff
@@ -82,29 +82,26 @@ public String getValue(String propertyName) {
     }
 
     private void prepareDataSource(final JdbcConfigConfig config) throws SQLException {
-        // create supplier
-        AgroalDataSourceConfigurationSupplier dataSourceConfiguration = new AgroalDataSourceConfigurationSupplier();
-        // get reference to connection pool
-        AgroalConnectionPoolConfigurationSupplier poolConfiguration = dataSourceConfiguration
-                .connectionPoolConfiguration();
-        // get reference to connection factory
-        AgroalConnectionFactoryConfigurationSupplier connectionFactoryConfiguration = poolConfiguration
-                .connectionFactoryConfiguration();
-
-        // configure pool
-        poolConfiguration
-                .initialSize(config.initialSize())
-                .minSize(config.minSize())
-                .maxSize(config.maxSize())
-                .acquisitionTimeout(config.acquisitionTimeout());
-
-        // configure supplier
-        connectionFactoryConfiguration
-                .jdbcUrl(config.url())
-                .credential(new NamePrincipal(config.username()))
-                .credential(new SimplePassword(config.password()));
-
-        dataSource = AgroalDataSource.from(dataSourceConfiguration.get());
+        var configuration = new AgroalDataSourceConfigurationSupplier()
+                .dataSourceImplementation(AGROAL)
+                .metricsEnabled(false)
+                .connectionPoolConfiguration(cp -> cp
+                        .minSize(config.minSize())
+                        .maxSize(config.maxSize())
+                        .initialSize(config.initialSize())
+                        .acquisitionTimeout(config.acquisitionTimeout())
+                        .connectionFactoryConfiguration(cf -> cf
+                                .jdbcUrl(config.url())
+                                .connectionProviderClassName(config.driver()
+                                        .orElse(null))
+                                .principal(config.username()
+                                        .map(NamePrincipal::new)
+                                        .orElse(null))
+                                .credential(config.password()
+                                        .map(SimplePassword::new)
+                                        .orElse(null))));
+
+        dataSource = AgroalDataSource.from(configuration);
```

Comment:
Can you please keep the old style and avoid lambdas and method references? Thanks :)